src/services.py

```diff
@@ -821,6 +822,19 @@ class ChatRoomService:
     def __init__(self, config):
         self.config = config
 
+    def archive_room(self, user_id, chat_room_id, is_archived):
```

Comment:
Does this delete the archive if it is archived?

I would advise not doing something like this. It seems like this is implementing the pattern where the front-end just fires some link, and the back-end will change accordingly. This is not so good for some reasons; for example, if somehow the front-end sends the request twice, there will be no change in the end, confusing the user.

I suggest having separate service functions `archive_room` and `unarchive_room`, as well as different API functions for both. The front-end can then send the corresponding API. If you make sure that the service functions are idempotent (https://stackoverflow.com/questions/1077412/what-is-an-idempotent-operation), it will be all good if say the front-end sends the function twice.

Reply:
ok noted. but the db is no issue?